src/main/java/com/palantir/common/streams/MoreStreams.java

```diff
@@ -57,6 +59,30 @@
     }
 
     /**
+     * Given a {@code Stream<ListenableFuture<U>>}, this function will return a blocking stream of the completed
```

Comment:
this function will return a blocking stream that waits for each future to complete before returning it, but which looks ahead `maxParallelism` futures to ensure a fixed parallelism rate

Reply:
sure